### PR TITLE
add coverage checks for Python 3.7-3.11

### PR DESCRIPTION
**Pull request by knsmith**
Addresses #165 for general-superstaq

### CODE REVIEW

**Comment by richrines1** [comment]
maybe a silly question, but is testing with python 3.7 and 3.10 sufficient to assume everything works w/ 3.8 and 3.9 as well?

**Comment by knsmith** [comment]
@richrines1 I'm not sure...I guess we were including all tests to ensure no surprises? Thoughts, @vtomole? We'll include Python 3.11 as well since it's been released.

**Review by vtomole — CHANGES_REQUESTED** [review]
Turns out we don't need to copy paste. We can have all the versions be in one check. See https://github.com/quantumlib/Cirq/blob/6fba8becd6d0fad05cfa7558ef5b1377b7a1d5b9/.github/workflows/ci.yml#L143

**Comment by knsmith** [comment]
This will look so much better! Yay!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -79,14 +79,17 @@ jobs:
           check/mypy_.py
 
   coverage:
-    name: Pytest and Coverage check
+    name: Pytest and Coverage check for Python 3.7 - 3.11
```
Comment (vtomole) [inline .github/workflows/ci.yml:82]: Lets revert this. The actions will show it anyway.
```suggestion
    name: Pytest and Coverage check
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -79,14 +79,17 @@ jobs:
...
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Set up Python 3.8
+      - name: Set up Python 3.7, 3.8, 3.9, 3.10, 3.11
```
Comment (vtomole) [inline .github/workflows/ci.yml:89]:
```suggestion
      - name: Set up Python
```

**Review by vtomole — CHANGES_REQUESTED** [review]

**Comment by knsmith** [comment]
@vtomole , did I fix all of your requested changes?

**Review by vtomole — APPROVED** [review]
Yep thank you so much!

**Review by vtomole — CHANGES_REQUESTED** [review]
Oh actually one more thing

**Comment by vtomole** [comment]
Change https://github.com/SupertechLabs/general-superstaq/blob/8f7c58a88f6c360f49002769fab921471d8bfeb2/setup.py#L38 from 3.6 to 3.7.

**Review by vtomole — APPROVED** [review]
LGTM

**Pull request merged**